src/bp/ui-studio/src/web/components/Layout/StatusBar/ConfigStatus.tsx

```diff
@@ -0,0 +1,100 @@
+import { Button, Classes, Colors, Dialog, Icon, Intent } from '@blueprintjs/core'
+import axios from 'axios'
+import React, { Fragment, useEffect, useState } from 'react'
+import { toastFailure } from '~/components/Shared/Utils'
+import EventBus from '~/util/EventBus'
+
+import ActionItem from './ActionItem'
+import style from './StatusBar.styl'
+
+const adminUrl = `${window['ROOT_PATH']}${window['API_PATH']}/admin/server/`
+
+const ConfigStatus = () => {
+  const [isDifferent, setDifferent] = useState(false)
+  const [isRestarting, setRestart] = useState(false)
+  const [isOpen, setOpen] = useState(false)
+
+  useEffect(() => {
+    // tslint:disable-next-line: no-floating-promises
+    fetchHash()
+
+    const configUpdated = event => setDifferent(event.initialHash !== event.newHash)
+
+    EventBus.default.on('config.updated', configUpdated)
+    return () => EventBus.default.off('config.updated', configUpdated)
+  }, [])
+
+  useEffect(() => {
+    if (!isRestarting) {
+      return
+    }
+
+    const interval = setInterval(async () => {
+      try {
+        await axios.get('/status', { timeout: 500 })
+        window.location.reload()
+      } catch (err) {} // silent catch intended
+    }, 1000)
+    return () => clearInterval(interval)
+  }, [isRestarting])
+
+  const fetchHash = async () => {
+    try {
+      const { data } = await axios.get(`${adminUrl}/configHash`)
+      if (data.initialHash && data.currentHash && data.initialHash !== data.currentHash) {
+        setDifferent(true)
+      }
+    } catch (err) {
+      toastFailure(err.message)
+    }
+  }
+
+  const restartServer = async () => {
+    try {
+      await axios.post(`${adminUrl}/rebootServer`)
+      setRestart(true)
+    } catch (err) {
+      toastFailure(err.message)
+    }
+  }
+
+  return (
+    <Fragment>
+      <ActionItem
+        id="statusbar_configstatus"
+        title="Config Status"
+        description="Pending changes"
+        className={style.right}
+        onClick={() => setOpen(true)}
+      >
+        {isDifferent && <Icon icon="cog" style={{ color: Colors.RED5 }} />}
```

Comment:
I suggest a warning icon instead.

Reply:
It's not really a warning, i'd keep it for warnings specially. Cog is meaningful since it's the config. Still open to discussion